crates/contracts/hydentity/src/main.rs

```diff
@@ -5,12 +5,23 @@ extern crate alloc;
 
 use hydentity::execute;
 use sdk::guest::{commit, GuestEnv, Risc0Env};
-use sdk::ContractInput;
+use sdk::{ContractInput, ProgramInput};
 
 risc0_zkvm::guest::entry!(main);
 
 fn main() {
     let env = Risc0Env {};
-    let input: ContractInput = env.read();
-    commit(env, input.clone(), execute(input));
+    let program_inputs: ProgramInput = env.read();
+    let hydentity_initial_state = borsh::from_slice(&program_inputs.serialized_initial_state)
+        .expect("Failed to decode state");
+    let res = execute(
+        hydentity_initial_state.clone(),
+        program_inputs.contract_input.clone(),
+    );
+    commit(
+        env,
+        hydentity_initial_state,
+        input.clone(),
+        execute(hydentity_initial_state, input),
+    );
```

Comment:
is it possible to keep the main as simple as it was and move this logic in the `execute` function ?

I have in mind a day where this file won't be necessary, so I would like to keep it as simple & always the same in all contracts